CMake/resolve_dependency_modules/README.md

```diff
@@ -1,7 +1,47 @@
+# Dependency List
+Following is the list of libraries and their minimum version
+that Velox requires. Some of these libraries can be installed
+via a platform's package manager (eg. `brew` on MacOS).
+The versions of certain libraries is the default provided by 
+the platform's package manager. Some libraries can be bundled
+by Velox. See details on bundling below.
+
+| Library Name      | Minimum Version | Bundled? | 
+|-------------------|-----------------|----------|
+| ninja             | default         | No       |
+| ccache            | default         | No       |
+| icu4c             | default         | Yes      |
+| gflags            | default         | yes      |
+| glog              | default         | Yes      |
+| gtest (testing)   | default         | Yes      |
+| libevent          | default         | No       |
+| lz4               | default         | No       | 
+| snappy            | default         | No       |
+| lzo               | default         | No       |
+| xz                | default         | No       |
+| zstd              | default         | No       |
+| openssl           | default         | No       |
+| protobuf          | 21 (exact)      | Yes      |
+| boost             | 1.66.0          | Yes      |
+| flex              | 2.5.13          | No       |
+| bison             | 3.0.4           | No       |
+| cmake             | 3.14            | No       |
+| double-conversion | 3.1.5           | No       |
+| xsimd             | 10.0.0          | Yes      |
```

Comment:
Does it mean that because the default value of `VELOX_DEPENDENCY_SOURCE` is `AUTO`, for a third-party library, if `set_source(xxx)` or `set(xxx_SOURCE AUTO)` is used in CMakeLists.txt (instead of `set(xxx_SOURCE BUNDLED)`), here it will be considered as `"bundled"`?

Unrelated to this PR, but there is an open question in #8308: Why does "xsimd" have to be `"bundled"`? Would love to know the answer, thanks.

Reply:
Because we have not made any modifications to "xsimd", I feel that we only need the specified "xsimd" version (for example: `find_package(xsimd 10.0.0 EXACT)`, or `resolve_dependency(xsimd 10.0.0 EXACT)`), but there is no need to require it to be compiled from source code, , that is, "xsimd" can be `set_source(xsimd)`, right?

Reply:
> Unrelated to this PR, but there is an open question in #8308: Why does "xsimd" have to be `"bundled"`? Would love to know the answer, thanks.

@lingbin I replied on that PR. You are right that the default is AUTO and can be overridden by `VELOX_DEPENDENCY_SOURCE`

Reply:
> that is, "xsimd" can be `set_source(xsimd)`, right?

This is right. Let's fix it here https://github.com/facebookincubator/velox/pull/8308

Reply:
Thanks for your reply, I will continue working on #8308